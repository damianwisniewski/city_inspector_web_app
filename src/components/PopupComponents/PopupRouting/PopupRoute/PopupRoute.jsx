import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { PopupContext } from '../PopupRouter/PopupRouter'
import Popup from '../../Popup/Popup'

class PopupRoute extends Component {
	static contextType = PopupContext

	render() {
		const { component, closeButton, darkOverlay } = this.props
		const { handleClosePopup, additionalProps } = this.context

		return (
			<Popup
				closeButton={closeButton}
				darkOverlay={darkOverlay}
				onOverlayClick={handleClosePopup}
				onButtonClose={handleClosePopup}
				{...additionalProps}
			>
				{React.createElement(component, { closePopup: handleClosePopup })}
			</Popup>
		)
	}
}

PopupRoute.defaultProps = {
	closeButton: false,
	darkOverlay: false,
}

PopupRoute.propTypes = {
	component: PropTypes.elementType.isRequired,
	closeButton: PropTypes.bool,
	darkOverlay: PropTypes.bool,
}

export default PopupRoute
