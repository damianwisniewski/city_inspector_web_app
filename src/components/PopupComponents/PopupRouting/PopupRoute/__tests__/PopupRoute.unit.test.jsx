import React from 'react'
import { configure, shallow, ShallowWrapper } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'

import PopupRoute from '../PopupRoute'

configure({ adapter: new Adapter() })

describe('<PopupRoute />', () => {
	/**
	 * @type {ShallowWrapper}
	 */
	let PopupRouteComponent
	const MockComponent = () => <p>mock</p>

	test('should render correctly', () => {
		PopupRouteComponent = shallow(<PopupRoute component={MockComponent} />)

		expect(toJson(PopupRouteComponent)).toMatchSnapshot()
	})

	test('should provide darkOverlay and closeButton to Popup', () => {
		PopupRouteComponent = shallow(<PopupRoute component={MockComponent} closeButton darkOverlay />)
		expect(PopupRouteComponent.find('Popup').props()).toMatchObject({
			closeButton: true,
			darkOverlay: true,
		})
	})

	//TODO: tests related to context
})
