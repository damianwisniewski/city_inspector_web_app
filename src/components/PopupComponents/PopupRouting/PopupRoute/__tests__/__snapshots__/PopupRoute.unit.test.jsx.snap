// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PopupRoute /> should render correctly 1`] = `
<Popup
  closeButton={false}
  darkOverlay={false}
>
  <MockComponent />
</Popup>
`;
